public/app/features/plugins/sql/utils/time.ts

```diff
@@ -0,0 +1,24 @@
+import { dateTime, rangeUtil, TimeRange } from '@grafana/data';
+import { TemplateSrv } from '@grafana/runtime';
+
+const RANGE_RE = /^(\d+) (\d+)$/; // find digits, then a space, then digits again.
+
+// there is no exported "getTimeSrv" functionality,
+// so we use the template server to access the current time range.
+export function getCurrentTimeRange(templateSrv: TemplateSrv): TimeRange {
+  const text = templateSrv.replace('$__from $__to');
```

Comment:
Explore updates TemplateSrv range only when a query is run, so `getCurrentTimeRange` may not get correct values, especially in split mode (though there are similar problem with TimeSrv). I was hoping we could correctly populate range in https://github.com/grafana/grafana/blob/main/public/app/features/query/components/QueryEditorRow.tsx#L294 so editors always get the correct time range as a prop.

Reply:
yeah, the situation is not perfect. all i can promise, is that this change will not make it more incorrect than before 😄 
(FYI:i'm looking into an alternative approach, see draft PR at https://github.com/grafana/grafana/pull/74800)